bitrepository-core/src/main/java/org/bitrepository/common/utils/TimeUtils.java

```diff
@@ -145,4 +188,32 @@ public static Date getMaxDate(Date currentMax, Date itemDate) {
             return currentMax;
         }
     }
+
+    /**
+     *
+     * @param duration a non-negative duration
+     * @return the duration converted to a long in either SECONDS, MILLISECONDS, MICROSECONDS or NANOSECONDS
+     * maintaining the best possible precision and truncated if necessary.
+     * @throws IllegalArgumentException if duration is negative
+     */
+    public static CountAndTimeUnit durationToCountAndTimeUnit(Duration duration) {
+        if (duration.isNegative()) {
```

Comment:
Wouldn't a switch case be prettier?

Reply:
I like the idea. However as far as I can see rewriting the conditions into something that would match constant case labels would introduce a complication that we don’t want. So immediately I think not, but feel free to suggest an advantageous way.